db/wal_manager.cc

```diff
@@ -321,9 +321,11 @@ Status WalManager::GetSortedWalsOfType(const std::string& path,
           s = env_->GetFileSize(archived_file, &size_bytes);
           if (!s.ok() && env_->FileExists(archived_file).IsNotFound()) {
```

Comment:
I am not sure I understand why we need to do an Exists, FileSize, Exists rather than just the FileSize.  The problem seems like there is a potential race between these three operations.  What is wrong with just FileSize and checking there if the file is found or not?

Reply:
I don't know. Base on the current logic, seems it could be file exist but FileSize unable to get, in such case it will return error. If we only check FileSize then it has to `continue` to skip that WAL file if FileSize failed.

Reply:
I am also curious to know if we can just use `GetFileSize()`, since we need to get its size anyway.

Reply:
There are two cases -- one for the wal_dir and one for the archive_dir.  The wal_dir uses GetFileSize directly.  It seems like the archive one could as well.

Reply:
make sense, switched to just use `GetFileSize()`.